### PR TITLE
Refactor Food Types CRUD

### PR DESCRIPTION
**Pull request by Fosome**
## Overview
The goal of this PR is to clean up the CRUD functionality around Food Types.

## Details
- Only region admins or super admins can CRUD Food Types, moving to the `region_admin` namespace
- Split each action into a separate interactor and test
- Feature test listing, creating, updating, and deleting Food Types in various roles

## Notes
This is probably too much testing around the feature specs - trying each operation as each role (visitor, volunteer, region admin, super admin).  I think it's worth adding now and removing once the app has stabilized more, maybe after the major Rails upgrades.

### CODE REVIEW

File: app/controllers/region_admin/food_types_controller.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
+class RegionAdmin::FoodTypesController < ApplicationController
+  before_filter :authenticate_volunteer!
+  before_filter :authorize_region_admin!
+
+  def index
+    @food_types = available_food_types
+
+    respond_to do |format|
+      format.html
+      format.json { render json: @food_types.to_json }
```
Comment (cllns) [inline app/controllers/region_admin/food_types_controller.rb]: I think you can get rid of `.to_json` :)

**Comment by rylanb** [comment]
I haven't researched code climate yet... what does '8 new issues' mean around this?

**Comment by cllns** [comment]
@rylanb click "Details" and it tells you :) In this case, it's alerting us that there are opportunities to DRY things up a bit. think we can tackle that afterwards, if we decide it's a good idea

File: app/controllers/region_admin/food_types_controller.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
+class RegionAdmin::FoodTypesController < ApplicationController
```
Comment (ericfreese) [inline app/controllers/region_admin/food_types_controller.rb]: module nesting here ;)

File: app/controllers/region_admin/food_types_controller.rb [hunk 3]
```diff
@@ -44,26 +37,63 @@ def create
...
+  end
+
+  private
+
+  def available_food_types
+    if current_volunteer.super_admin?
+      FoodType.active
+    else
+      FoodType.active.regional(current_volunteer.region_ids)
+    end
```
Comment (ericfreese) [inline app/controllers/region_admin/food_types_controller.rb]: What do you think of using CanCan's `FoodType.accessible_by(current_ability)` helper here?

File: app/controllers/region_admin/food_types_controller.rb [hunk 4]
```diff
@@ -44,26 +37,63 @@ def create
...
+  end
+
+  def available_regions
+    if current_volunteer.super_admin?
+      Region.all
+    else
+      current_volunteer.assignments.collect{ |a| a.admin ? a.region : nil }.compact
+    end
```
Comment (ericfreese) [inline app/controllers/region_admin/food_types_controller.rb]: You may be able to use `Region.accessible_by(current_ability)` here as well. Not entirely sure though.

File: app/controllers/region_admin/food_types_controller.rb [hunk 5]
```diff
@@ -44,26 +37,63 @@ def create
...
+  end
+
+  def authorize_region_admin!
+    return if region_admin?
+    redirect_to root_url, alert: "Unauthorized"
+  end
+
+  def region_admin?
+    current_volunteer.any_admin?
```
Comment (ericfreese) [inline app/controllers/region_admin/food_types_controller.rb]: How about changing `Ability#region_admin?` to public and delegating to that here to reduce duplicated logic?

File: app/controllers/region_admin/food_types_controller.rb [hunk 6]
```diff
@@ -44,26 +37,63 @@ def create
...
+    if current_volunteer.super_admin?
+      Region.all
+    else
+      current_volunteer.assignments.collect{ |a| a.admin ? a.region : nil }.compact
+    end
+  end
+
+  def authorize_region_admin!
+    return if region_admin?
+    redirect_to root_url, alert: "Unauthorized"
```
Comment (ericfreese) [inline app/controllers/region_admin/food_types_controller.rb]: We could raise a `CanCan::AccessDenied` error here to maintain consistent behavior when unauthorized

File: app/controllers/region_admin/food_types_controller.rb (hunk 4 above)
Comment (Fosome) [inline app/controllers/region_admin/food_types_controller.rb]: Using `cancancan` AR extensions has been leaving me with bad feels.  I don't like the idea of using the gem to handle DB lookups - it's too mysterious to me.

Counter feels?

**Comment by rylanb** [comment]
@Fosome what is the status of this? Can we get it ready to merge and squashed down commit wise?

**Pull request merged**